homeassistant/components/qnap/sensor.py

```diff
@@ -1,346 +1,323 @@
-ATTR_MEMORY_SIZE = "Memory Size"
-ATTR_MODEL = "Model"
-ATTR_PACKETS_TX = "Packets (TX)"
-ATTR_PACKETS_RX = "Packets (RX)"
-ATTR_PACKETS_ERR = "Packets (Err)"
-ATTR_SERIAL = "Serial #"
-ATTR_TYPE = "Type"
-ATTR_UPTIME = "Uptime"
-ATTR_VOLUME_SIZE = "Volume Size"
-
-CONF_DRIVES = "drives"
-CONF_NICS = "nics"
-CONF_VOLUMES = "volumes"
-DEFAULT_NAME = "QNAP"
-DEFAULT_PORT = 8080
-DEFAULT_TIMEOUT = 5
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
-
-NOTIFICATION_ID = "qnap_notification"
-NOTIFICATION_TITLE = "QNAP Sensor Setup"
-
-_SYSTEM_MON_COND: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigType,
+    async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    uid = config_entry.unique_id
+    sensors: list[QNAPSensor] = []
+
+    sensors.extend(
+        [QNAPSystemSensor(coordinator, description, uid) for description in BAS_SENSOR]
+    )
+
+    sensors.extend(
+        [QNAPCPUSensor(coordinator, description, uid) for description in CPU_SENSOR]
+    )
+
+    sensors.extend(
+        [QNAPMemorySensor(coordinator, description, uid) for description in MEM_SENSOR]
+    )
+
+    # Network sensors
+    sensors.extend(
+        [
+            QNAPNetworkSensor(coordinator, description, uid, nic)
+            for nic in coordinator.data["system_stats"]["nics"].keys()
+            for description in NET_SENSOR
+        ]
+    )
+
+    # Drive sensors
+    sensors.extend(
+        [
+            QNAPDriveSensor(coordinator, description, uid, drive)
+            for drive in coordinator.data["smart_drive_health"].keys()
+            for description in DRI_SENSOR
+        ]
+    )
+
+    # Volume sensors
+    sensors.extend(
+        [
+            QNAPVolumeSensor(coordinator, description, uid, volume)
+            for volume in coordinator.data["volumes"].keys()
+            for description in VOL_SENSOR
+        ]
+    )
+
+    # Folders sensors
+    sensors.extend(
+        [
+            QNAPFolderSensor(coordinator, description, uid, volume, folder["sharename"])
+            for volume in coordinator.data["volumes"].keys()
+            for folder in coordinator.data["volumes"][volume].get("folders", [])
+            for description in FOL_SENSOR
+        ]
+    )
+    async_add_entities(sensors)
+
+
+def round_nicely(number):
+    """Round a number based on its size (so it looks nice)."""
+    if number < 10:
+        return round(number, 2)
+    if number < 100:
+        return round(number, 1)
+
+    return round(number)
+
+@dataclass
+class QNapSensorEntityDescription(SensorEntityDescription):
+    """Represents an Flow Sensor."""
+
+    stype: str | None = None
+
+
+SENSOR_TYPES: tuple[QNapSensorEntityDescription, ...] = (
+    QNapSensorEntityDescription(
+        stype="basic",
         key="status",
-        name="Status",
+        name="Health",
```

Comment:
Changing the name, the icon or the state class of the sensors has nothing to do with config-flow.
I suggest that you create a separate PR for the entity description amends.

Also, entity_registry_enabled_default defaults to True so you only need to set it explicitely when it's False.

Reply:
Reverted name and removed all entity_registry_enabled_default set to true